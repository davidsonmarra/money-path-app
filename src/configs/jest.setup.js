import '@testing-library/react-native/extend-expect';
import {mockGoogleSignIn} from 'src/__mocks__';

jest.useFakeTimers();

const mockTheme = {
  colors: {
    background: '#232E38',
    primary: '#EAC43D',
    secondary: '#2A5C99',
    info: '#14C0CC',
    success: '#A6CD4E',
    title: '#FFF',
    text: '#D9D9D9',
    disabled: '#4F4F4F',
    buttonText: '#FFF',
  },
};

jest.mock('src/hooks/useTheme', () => ({
  useTheme: () => ({
    theme: mockTheme,
    dark: true,
  }),
}));

jest.mock('@react-native-firebase/auth', () => ({
  firebase: {
    auth: () => ({
      signInWithCredential: jest.fn(),
      signOut: jest.fn(),
    }),
  },
  GoogleAuthProvider: {
    credential: jest.fn(),
  },
}));

jest.mock('@react-native-google-signin/google-signin', () => ({
  GoogleSignin: mockGoogleSignIn,
}));

jest.mock('react-native-safe-area-context', () => ({
  ...jest.requireActual('react-native-safe-area-context'),
  useSafeAreaInsets: jest.fn(() => ({top: 0, bottom: 0, left: 0, right: 0})),
}));
