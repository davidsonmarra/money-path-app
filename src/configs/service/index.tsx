import axios from 'axios';

const api = axios.create({
  baseURL: 'https://moneypathapi-production.up.railway.app',
  timeout: 10000,
  headers: {
    Authorization: `eyJhbGciOiJSUzI1NiIsImtpZCI6IjgxYjUyMjFlN2E1ZGUwZTVhZjQ5N2UzNzVhNzRiMDZkODJiYTc4OGIiLCJ0eXAiOiJKV1QifQ.eyJuYW1lIjoiRGF2aWRzb24gTWFycmEgUm9kcmlndWVzIFZpZWlyYSIsInBpY3R1cmUiOiJodHRwczovL2xoMy5nb29nbGV1c2VyY29udGVudC5jb20vYS9BQ2c4b2NJNWtza25jVkozNVJ5aWFxbk1NcmRlWmtkRjFkQ1g3OXlhTHE2dVBORThMMlotMHdNPXM5Ni1jIiwiaXNzIjoiaHR0cHM6Ly9zZWN1cmV0b2tlbi5nb29nbGUuY29tL21vbmV5LXBhdGgtZHAiLCJhdWQiOiJtb25leS1wYXRoLWRwIiwiYXV0aF90aW1lIjoxNzM4MDI2OTg5LCJ1c2VyX2lkIjoiVTFZWm9HRTFWSk5FcTJPOFdpSk83WXhnWW42MyIsInN1YiI6IlUxWVpvR0UxVkpORXEyTzhXaUpPN1l4Z1luNjMiLCJpYXQiOjE3MzgwMjY5ODksImV4cCI6MTczODAzMDU4OSwiZW1haWwiOiJkYXZpZHNvbi52aWVpcmFAc291cGlsYXIuY29tLmJyIiwiZW1haWxfdmVyaWZpZWQiOnRydWUsImZpcmViYXNlIjp7ImlkZW50aXRpZXMiOnsiZ29vZ2xlLmNvbSI6WyIxMTQ4OTYzMjk2MzkxMzQ2MzQyNzYiXSwiZW1haWwiOlsiZGF2aWRzb24udmllaXJhQHNvdXBpbGFyLmNvbS5iciJdfSwic2lnbl9pbl9wcm92aWRlciI6Imdvb2dsZS5jb20ifX0.K80hDFbF5FqABuWtlRfcvaSRintNMPiMY6unnRkXo4sCX32OQIL2JAw-7bJPYtvM7QqiMnbbVxVf0YU68EBcJ5ydA8hRChr_Hkitd1nDIDcf8uC4hmuH-03PLXG0VjLGbMLxE9XAdal51oaaiRa0MxT1_4jvuhSNQsQn22YdvBBEiF8-NqfAGCcwDlHoO5sXtYBkQJA-iK_KPAUPcOC2zSF1iLOZChpFhhq1dZKSHX_WsO-QiC0pn38vTgp801mGookgBcZt8FnQCVlol4NWN3sUWL38Vlx73h-Ts2pxU9MqO94OJss19xsZXpOoAAEgMflsfERfvtuOrzVGdYgLTg`,
  },
});

export default api;
