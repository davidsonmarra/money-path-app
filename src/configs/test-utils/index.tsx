import {render} from '@testing-library/react-native';
import {SafeAreaView} from 'react-native-safe-area-context';
import AddWalletFormContext from 'src/features/add-wallet/context/add-wallet-form';

interface IAllThemeProviders {
  children: React.ReactElement;
}

const AllTheProviders = ({children}: IAllThemeProviders) => (
  <AddWalletFormContext>
    <SafeAreaView>{children}</SafeAreaView>
  </AddWalletFormContext>
);

const customRender = (ui: React.ReactElement, options = {}) =>
  render(ui, {wrapper: AllTheProviders, ...options});

export const renderWithCustomProviders = (component: React.ReactElement) => {
  return {
    render: render(<SafeAreaView>{component}</SafeAreaView>),
  };
};

// re-export everything
export * from '@testing-library/react-native';

// override render method
export {customRender as render};
