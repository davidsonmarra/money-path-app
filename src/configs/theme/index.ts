export {default as colors} from 'src/configs/theme/colors';
