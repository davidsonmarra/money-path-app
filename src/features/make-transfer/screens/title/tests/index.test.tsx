import React from 'react';
import { fireEvent, render } from 'src/configs/test-utils';
import TitleScreen from 'src/features/make-transfer/screens/title';

const mockNavigate = jest.fn();
jest.mock('@react-navigation/native', () => ({
  useNavigation: () => ({
    goBack: jest.fn(),
    navigate: mockNavigate,
  }),
}));

describe('TitleScreen', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly with default props', () => {
    const screen = render(<TitleScreen />).toJSON();
    expect(screen).toMatchSnapshot();
  });

  it('should pass correct props to TitleContainer', () => {
    const { getByText, getByTestId } = render(<TitleScreen />);

    expect(getByText('Digite o título')).toBeTruthy();
    expect(getByText('Descrição (opcional):')).toBeTruthy();
    expect(getByTestId('input-title')).toBeTruthy();
    expect(getByTestId('input-description')).toBeTruthy();
    expect(getByTestId('btn-confirm')).toBeTruthy();
  });

  it('should navigate to AddWalletStack when onConfirm is called', () => {
    const { getByTestId } = render(<TitleScreen />);

    const input = getByTestId('input-title');
    const confirmButton = getByTestId('btn-confirm');

    fireEvent.changeText(input, 'Teste de título');
    fireEvent.press(confirmButton);

    expect(mockNavigate).toHaveBeenCalledWith('MakeTransferStack', {
      screen: 'Type',
    });
  });

  it('should disable confirm button when title is empty', () => {
    const { getByTestId } = render(<TitleScreen />);

    const confirmButton = getByTestId('btn-confirm');
    expect(confirmButton.props.accessibilityState.disabled).toBe(true);
  });

  it('should enable confirm button when title is not empty', () => {
    const { getByTestId } = render(<TitleScreen />);

    const input = getByTestId('input-title');
    const confirmButton = getByTestId('btn-confirm');

    fireEvent.changeText(input, 'Teste de título');

    expect(confirmButton.props.accessibilityState.disabled).toBe(false);
  });
});
