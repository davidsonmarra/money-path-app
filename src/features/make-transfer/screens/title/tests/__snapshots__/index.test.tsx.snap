// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleScreen should render correctly with default props 1`] = `
<RNCSafeAreaView
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#2A5C99",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingBottom": 12,
        "paddingHorizontal": 16,
        "paddingTop": 12,
        "width": "100%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="btn-left-icon"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={24}
        bbWidth={24}
        fill="#FFF"
        focusable={false}
        height={24}
        meetOrSlice={0}
        minX={0}
        minY={-960}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {
              "flex": 0,
              "height": 24,
              "width": 24,
            },
          ]
        }
        testID="icon-back"
        vbHeight={960}
        vbWidth={960}
        width={24}
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4294967295,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M360-240 120-480l240-240 56 56-144 144h568v80H272l144 144z"
            fill={
              {
                "payload": 4278190080,
                "type": 0,
              }
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
    <Text
      style={
        [
          {
            "color": "#D9D9D9",
            "fontSize": 16,
            "fontWeight": 600,
            "lineHeight": 16,
          },
          {
            "color": "#FFF",
          },
        ]
      }
    >
      Título
    </Text>
    <View />
  </View>
  <View
    onLayout={[Function]}
    style={
      [
        {
          "flex": 1,
        },
        {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <RNCSafeAreaView
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessible={true}
      edges={
        {
          "bottom": "additive",
          "left": "off",
          "right": "off",
          "top": "off",
        }
      }
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#232E38",
          "flex": 1,
          "justifyContent": "space-between",
          "paddingHorizontal": 24,
          "paddingVertical": 16,
        }
      }
    >
      <View>
        <Text
          style={
            [
              {
                "color": "#D9D9D9",
                "fontSize": 20,
                "fontWeight": 600,
                "lineHeight": 24,
              },
              undefined,
            ]
          }
        >
          Digite o título
        </Text>
        <View
          style={
            {
              "height": 12,
              "width": 0,
            }
          }
          testID="spacer"
        />
        <Text
          style={
            [
              [
                {
                  "color": "#D9D9D9",
                  "fontSize": 14,
                  "lineHeight": 20,
                },
                {
                  "fontWeight": 400,
                },
              ],
              undefined,
            ]
          }
        >
          Para criar uma transferência, precisamos que você digite um título para identificá-la:
        </Text>
        <View
          style={
            {
              "height": 24,
              "width": 0,
            }
          }
          testID="spacer"
        />
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#232E38",
              "borderColor": "#FFFFFF33",
              "borderRadius": 8,
              "borderWidth": 1,
              "flexDirection": "row",
              "paddingHorizontal": 16,
              "width": "100%",
            }
          }
        >
          <TextInput
            editable={true}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder="Ex: Pagamento da conta de luz"
            placeholderTextColor="#FFFFFF65"
            selectTextOnFocus={true}
            style={
              {
                "color": "#D9D9D9",
                "paddingVertical": 12,
                "width": "100%",
              }
            }
            testID="input-title"
            value=""
          />
        </View>
      </View>
      <View>
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": true,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#4F4F4F",
              "borderRadius": 12,
              "flexDirection": "row",
              "height": 48,
              "justifyContent": "space-between",
              "opacity": 1,
              "paddingHorizontal": 24,
              "paddingVertical": 12,
              "width": "100%",
            }
          }
          testID="btn-confirm"
        >
          <View />
          <Text
            style={
              [
                [
                  {
                    "color": "#D9D9D9",
                    "fontSize": 16,
                    "lineHeight": 24,
                  },
                  {
                    "fontWeight": 600,
                  },
                ],
                {
                  "color": "#FFF",
                },
              ]
            }
          >
            Continuar
          </Text>
          <View />
        </View>
      </View>
    </RNCSafeAreaView>
  </View>
</RNCSafeAreaView>
`;
